### PR TITLE
[CI] Drop DPCPP_ENABLE_OPAQUE_POINTERS cmake flag

### PR DESCRIPTION
**Pull request by bader**
All tools are switched to opaque pointers already except those we use in compilation for offload targets.

### CODE REVIEW

File: .github/workflows/sycl_nightly.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ jobs:
       build_cache_root: "/__w/"
       build_cache_suffix: opaque_pointers
       build_artifact_suffix: opaque_pointers
-      build_configure_extra_args: "--hip --cuda --enable-esimd-emulator --cmake-opt=-DDPCPP_ENABLE_OPAQUE_POINTERS=TRUE --cmake-opt=-DSPIRV_ENABLE_OPAQUE_POINTERS=TRUE"
+      build_configure_extra_args: "--hip --cuda --enable-esimd-emulator --cmake-opt=-DSPIRV_ENABLE_OPAQUE_POINTERS=TRUE"
```
Comment (bader) [inline .github/workflows/sycl_nightly.yml:50]: @jcranmer-intel, `DPCPP_ENABLE_OPAQUE_POINTERS=TRUE` should be a dead code now. Right?

Comment (bader) [reply .github/workflows/sycl_nightly.yml:50]: Sorry I used wrong wording. 
I tested it locally and I see that `SPIRV_ENABLE_OPAQUE_POINTERS=TRUE` is able to expose all the issues with opaque pointers as `DPCPP_ENABLE_OPAQUE_POINTERS=TRUE` does.

Comment (jcranmer-intel) [reply .github/workflows/sycl_nightly.yml:50]: The default of `DPCPP_ENABLE_OPAQUE_POINTERS` should be flipped to true for `SPIRV_ENABLE_OPAQUE_POINTERS` to have meaning. At this point, it's probably possible to flip the default in cmake of `DPCPP_ENABLE_OPAQUE_POINTERS`, given that most of the current failures should have been moved to requiring `SPIRV_ENABLE_OPAQUE_POINTERS` to be true.

Comment (bader) [reply .github/workflows/sycl_nightly.yml:50]: > At this point, it's probably possible to flip the default in cmake of `DPCPP_ENABLE_OPAQUE_POINTERS`, given that most of the current failures should have been moved to requiring `SPIRV_ENABLE_OPAQUE_POINTERS` to be true.

CI confirms your guess. There is one failing LIT test there and we probably can fix it by updating the checks (they are looking for a typed pointer).

Comment (bader) [reply .github/workflows/sycl_nightly.yml:50]: Please, disregard my previous comment. I mixed up flags.
I enabled `DPCPP_ENABLE_OPAQUE_POINTERS` by default, but llvm-spirv tests are failing and there are tests in `check-sycl` crashing at SPIR-V translation.
```
Failed Tests (2):
  SYCL :: matrix/legacy/matrix-elemwise-ops.cpp
  SYCL :: matrix/matrix-elemwise-ops.cpp
```

**Review by jchlanda — APPROVED** [review]
libclc/remangler LGTM

**Review by AlexeySachkov — APPROVED** [review]

**Review by jcranmer-intel — APPROVED** [review]

**Comment by bader** [comment]
@intel/dpcpp-devops-reviewers, ping.

**Review by aelovikov-intel — APPROVED** [review]

**Pull request merged**